artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/LinkedListImpl.java

```diff
@@ -203,20 +223,31 @@ public void addSorted(E e) {
          // This would be an optimization for our usage.
          // avoiding scanning the entire List just to add at the end, so we compare the end first.
          if (comparator.compare(tail.val(), e) >= 0) {
+            logger.trace("addTail as compare is after tail {}", e);
```

Comment:
Perhaps log the tail value as well as e? We can already know what e is when reading the logs from the earlier logging by the thread, but we wont necessarily know what tail was when the comparison was made.